plugins/uefi-capsule/fu-uefi-backend-linux.c

```diff
@@ -70,15 +76,53 @@ fu_uefi_backend_device_new (const gchar *path)
 	return g_steal_pointer (&dev);
 }
 
+void
+fu_uefi_backend_set_enable_cod (FuBackend *backend, gboolean enable_cod)
+{
+	FuUefiBackend *self = FU_UEFI_BACKEND (backend);
+	self->enable_cod = !enable_cod;
+}
+
+static gboolean
+fu_backend_uefi_check_cod_support (FuUefiBackend *self, GError **error)
+{
+	gsize bufsz = 0;
+	guint64 value = 0;
+	g_autofree guint8 *buf = NULL;
+
+	if (!fu_efivar_get_data (FU_EFIVAR_GUID_EFI_GLOBAL, "OsIndicationsSupported",
+				 &buf, &bufsz, NULL, error)) {
+		g_prefix_error (error, "failed to read EFI variable: ");
+		return FALSE;
+	}
+	if (!fu_memcpy_safe ((guint8 *) &value, sizeof(value), 0x0,	/* dst */
+			      buf, bufsz, 0x0, sizeof(value), error))
+		return FALSE;
```

Comment:
Why do we need to copy the value here? Can't we just use buf?

Reply:
I was wrong, memcpy isn't endian-clean. Fixed.